src/rag/vector_store/pgvector.py

```
@@ -1,28 +1,347 @@
+                "PGVECTOR_DSN environment variable or dsn parameter is required"
+            )
+
+    def _get_pool(self):
+        """커넥션 풀을 lazy하게 초기화."""
+        if self._pool is None:
+            try:
+                from psycopg_pool import ConnectionPool
+            except ImportError:
+                raise ImportError(
+                    "psycopg[pool] is required. Install with: pip install 'psycopg[binary,pool]'"
+                )
```

Ruff (0.15.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)